File: network_config.py
"""
Network Configuration untuk CurtainCall Backend
Semua konfigurasi IP address dan network settings ada di sini
"""

import os
from dotenv import load_dotenv

load_dotenv()

class NetworkConfig:
    """Konfigurasi network untuk backend"""
    
    # MQTT Broker Configuration
    MQTT_BROKER_IP = os.getenv('MQTT_BROKER_IP', '127.0.0.1')
    MQTT_BROKER_PORT = int(os.getenv('MQTT_BROKER_PORT', 1883))
    MQTT_KEEPALIVE = int(os.getenv('MQTT_KEEPALIVE', 60))
    MQTT_TLS_ENABLED = os.getenv('MQTT_TLS_ENABLED', 'False').lower() == 'true'
    
    # Flask Server Configuration
    FLASK_HOST = os.getenv('FLASK_HOST', '0.0.0.0')
    FLASK_PORT = int(os.getenv('FLASK_PORT', 5000))
    
    # MongoDB Configuration
    MONGO_HOST = os.getenv('MONGO_HOST', 'localhost')
    MONGO_PORT = int(os.getenv('MONGO_PORT', 27017))
    MONGO_DB_NAME = os.getenv('MONGO_DB_NAME', 'curtaincall')
    
    @classmethod
    def get_mqtt_broker_url(cls):
        """Get full MQTT broker URL"""
        return cls.MQTT_BROKER_IP
    
    @classmethod
    def get_mongo_uri(cls):
        """Get MongoDB connection URI"""
        return f"mongodb://{cls.MONGO_HOST}:{cls.MONGO_PORT}/{cls.MONGO_DB_NAME}"
    
    @classmethod
    def print_config(cls):
        """Print current network configuration"""
        print("=" * 50)
        print("Network Configuration")
        print("=" * 50)
        print(f"MQTT Broker: {cls.MQTT_BROKER_IP}:{cls.MQTT_BROKER_PORT}")
        print(f"Flask Server: {cls.FLASK_HOST}:{cls.FLASK_PORT}")
        print(f"MongoDB: {cls.MONGO_HOST}:{cls.MONGO_PORT}/{cls.MONGO_DB_NAME}")
        print("=" * 50)

File: app/config/network.config.ts
import { Platform } from 'react-native';

/**
 * Network Configuration untuk CurtainCall Frontend
 * Semua konfigurasi IP address dan network settings ada di sini
 */

// ⚙️ KONFIGURASI IP ADDRESS
// Ganti dengan IP address komputer/server Anda
const BACKEND_IP = '192.168.1.5'; // IP address backend Flask
const BACKEND_PORT = 5000; // Port backend Flask
const MQTT_BROKER_IP = '192.168.1.5'; // IP address MQTT broker
const MQTT_BROKER_PORT = 1883; // Port MQTT broker

// ⚙️ FORCE PHYSICAL DEVICE MODE
// Set ke true jika menggunakan physical device (Android/iOS)
// Set ke false untuk menggunakan auto-detect (emulator/simulator)
const FORCE_PHYSICAL_DEVICE = true; // ⚙️ UBAH INI JIKA PERLU

/**
 * Get API Base URL berdasarkan platform
 */
const getBaseURL = () => {
  if (__DEV__) {
    // Jika force physical device mode, selalu gunakan custom IP
    if (FORCE_PHYSICAL_DEVICE) {
      console.log('📱 Using physical device mode with IP:', BACKEND_IP);
      return `http://${BACKEND_IP}:${BACKEND_PORT}/api`;
    }
    
    // Auto-detect berdasarkan platform
    if (Platform.OS === 'android') {
      // Android emulator menggunakan 10.0.2.2 untuk mengakses host machine
      return `http://10.0.2.2:${BACKEND_PORT}/api`;
    } else if (Platform.OS === 'ios') {
      // iOS simulator bisa pakai localhost
      return `http://localhost:${BACKEND_PORT}/api`;
    } else {
      // Web atau platform lain
      return `http://localhost:${BACKEND_PORT}/api`;
    }
  }
  // Production mode - gunakan IP yang dikonfigurasi
  return `http://${BACKEND_IP}:${BACKEND_PORT}/api`;
};

/**
 * Network Configuration
 */
export const NETWORK_CONFIG = {
  // Backend API Configuration
  BACKEND_IP,
  BACKEND_PORT,
  API_BASE_URL: getBaseURL(),
  API_TIMEOUT: 30000, // 30 detik
  
  // MQTT Broker Configuration (untuk referensi jika diperlukan di frontend)
  MQTT_BROKER_IP,
  MQTT_BROKER_PORT,
  
  // Platform info
  PLATFORM: Platform.OS,
};

/**
 * Helper function untuk mendapatkan full API URL
 */
export const getApiUrl = (endpoint: string): string => {
  const baseUrl = NETWORK_CONFIG.API_BASE_URL;
  // Remove leading slash jika ada
  const cleanEndpoint = endpoint.startsWith('/') ? endpoint.substring(1) : endpoint;
  return `${baseUrl}/${cleanEndpoint}`;
};

export default NETWORK_CONFIG;

